sql/insert.go

```diff
@@ -77,8 +77,16 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 
 	for rows.Next() {
 		values := rows.Values()
-		if len(values) != len(cols) {
-			return nil, fmt.Errorf("invalid values for columns: %d != %d", len(values), len(cols))
+		for range cols[len(values):] {
+			values = append(values, parser.DNull)
+		}
+
+		for _, col := range tableDesc.Columns {
+			if !col.Nullable {
+				if i, ok := colIDtoRowIndex[col.ID]; !ok || values[i] == parser.DNull {
```

Comment:
I think it only matters for performance, but we could check that all not-null columns are being inserted into before the table scan. See lines 59-64 where we do something similar for verifying all of the primary-key columns are being inserted into.